package.json

```diff
@@ -20,6 +20,7 @@
     "discord.js": "^14.15.3",
     "got": "^11.8.2",
     "jpeg-js": "^0.3.7",
+    "node-schedule": "^2.1.1",
```

Comment:
`node-schedule` is specifically designed for scheduling within a running process. It explicitly does not support scheduled tasks which run when the process is not running, and it explicitly does not persist tasks between process starts. If the Chubby process were to need to be restarted, we would lose all scheduled tasks

`node-schedule` mentions this in it's README with a list of alternatives:

> Note that Node Schedule is designed for in-process scheduling, i.e. scheduled jobs will only fire as long as your script is running, and the schedule will disappear when execution completes. If you need to schedule jobs that will persist even when your script isn't running, consider using actual [cron](http://unixhelp.ed.ac.uk/CGI/man-cgi?crontab+5).
> 
> In case you need durable jobs that persist across restarts and lock system compatible with multi-node deployments, try [agenda](https://github.com/agenda/agenda) or [bree](https://github.com/breejs/bree).

This is not suitable as-is and would need to be changed before this can be merged

Reply:
> If the Chubby process were to need to be restarted, we would lose all scheduled tasks.

I don't see why persisting tasks between process starts is necessary. All state is stored in the `matchmaking_threads` table and the schedule only calls `checkMatchmakingThreads`, which checks the `matchmaking_threads` table to close threads that haven't had any recent activity. I don't think losing scheduled tasks would have any negative effects besides delaying some thread closures.

As far as I can tell, Chubby downtime would cause the following effects:

1. Some messages in threads would be missed, resulting in the `last_message_sent` for those threads not being updated. This issue fixes itself the next time someone sends a message in that thread. Even if an entire thread gets missed, the entry would still be created the next time a message is sent.
2. Thread closure would be delayed by the downtime + 30 minutes. 30 was just chosen because it seemed reasonable, this could definitely be reduced for better accuracy. I would not be concerned about this because it is just a delay. As long as messages are sent in the thread after Chubby goes back up, no threads are missed.

Reply:
From what I can tell `checkMatchmakingThreads` is only called inside the scheduled task. There is no self recovery at start or anything like that. Which means that if the bot restarts, we lose those tasks and `checkMatchmakingThreads` will only be called once a *new* task is made in the new process instance

Reply:
Yes, it's one global task, so I believe `node-schedule` should be okay in that context. The task could be run more often, I didn't choose 30 minutes for any particular reason.

Reply:
I think the best resolution to this would be to check the last message of every thread in `checkMatchmakingThreads` before closing it, just to be safe. Then I believe it would be safe to always run it on start.